### PR TITLE
Added schema.registry.url propery to support Confluent schema registry

### PR DESCRIPTION
**Pull request by kcturner123**

### CODE REVIEW

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConfiguration.java [hunk 1]
```diff
@@ -775,6 +779,20 @@ public void setBrokers(String brokers) {
...
+    public String getSchemaRegistryURL() {
+		return schemaRegistryURL;
+	}
+
+    /**
+     * URL of the Kafka schema registry to use.
+     * The format is host1:port1,host2:port2.
+     * <p/>
+     * This option is known as <tt>schema.registry.url</tt> in the Kafka documentation.
+     */
```
Comment (oscerd) [inline components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConfiguration.java]: I think it must be stressed that this something related to Confluent only. There is no trace of schema.registry.url in the official Kafka documentation. So I think the javadoc must be changed. Also you need to run a build of the component so the associated .adoc file will be aligned to the new parameter.

Comment (davsclaus) [reply components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConfiguration.java]: Yeah that must be documented that its confluent only.

**Comment by davsclaus** [comment]
Instead of chasing all the extra options from confluent, then maybe we should have a generic way of configuring extra options, then the user can set them that way.

We have other components that does that with a prefix, eg `kafka...?extra.foo=123&extra.bar=blah` where `extra.` is the prefix. We can come up with a better name than just `extra`

**Comment by oscerd** [comment]
Yes, this can be part of a different issue, maybe we can merge this option and then move that to the new extra.** model, since we will release 3.0 in the future.

**Comment by oscerd** [comment]
@kcturner123 please update the javadoc accordingly. Thanks

**Comment by oscerd** [comment]
Thanks, this has been merged on master.

**Comment by bozic** [comment]
@kcturner123 can you provide a working example for this? We have Confluent Kafka 5.0.0.  and I can't connect consume messages. I did build locally this commit, but I have 2 issues:
* on some topics the consumer just failes when doing offset commit - similar to [this bug](https://issues.apache.org/jira/browse/CAMEL-12031)
* I can't get the deserialized object down to next step in route because for some reason, camel route expect to get the converter from the my Avro class (TestOrder) to InputStream? Is that usual behavior in phase of unmarshaling? And when I create a simple converter from my class to ByteArrayInputStream I get EOF Exception thru somewhere from avro package.... 

Camel route: https://hastebin.com/rejesiqere.java
Project POM: https://hastebin.com/lijusufeci.xml

There is also missing the `specific.avro.reader` property if you want to deserialize in class and probably a bunch of others, so it could be better idea with extra.** model.

**Pull request closed without merging**